src/telegraf.ts

```diff
@@ -78,6 +77,8 @@ export class Telegraf<
 > extends Composer<TContext> {
   private readonly options: Telegraf.Options<TContext>
   private webhookServer?: http.Server | https.Server
+  /** Set manually to avoid implicit `getMe` call in `launch` or `webhookCallback` */
+  public botInfo!: tt.UserFromGetMe
```

Comment:
How about simply `throw`ing in `handleUpdate` if `botInfo` is not set?

```suggestion
  public botInfo?: tt.UserFromGetMe
```

Reply:
The whole point of this PR is to make botInfo required. Otherwise, library consumers have to check for `botInfo` being absent and request it themselves, which is inconvenient.

Also, none of our tests sets `botInfo`, so we'd have to fix all of those now …

Reply:
`Context::botInfo` should be required, not necessarily `Telegraf::botInfo`.

Reply:
How are you going to initialize `Context` from `Telegraf` if `Telegraf` might not have access to `botInfo`?

Reply:
This would fail if `Telegraf::botInfo` is optional:

https://github.com/telegraf/telegraf/pull/1200/files#diff-407ce1c4de4ad023ddb3930497b90b8c0ca458d9a6096cea80abe06692d250c9R256

Reply:
Not if you throw in `handleUpdate` if it's not set

Reply:
Mind if I try?

Reply:
It would compile, that it true. All respective unit tests would fail, though. Feel free to set `botInfo` everywhere manually, I don't think its worth the effort.

Why is it even that you want to allow setting `botInfo` to `undefined` if every new request would try initialize it again anyway?

Reply:
> Mind if I try?

Of course not!

Reply:
> All respective unit tests would fail, though.

Oh, I see. Then it's fine as-is.